loopy/schedule/tree.py

```diff
@@ -110,42 +110,36 @@ def parent(self, node: NodeT) -> NodeT | None:
         """
         Returns the parent of *node*.
         """
-        assert node in self
-
         return self._child_to_parent[node]
 
     def children(self, node: NodeT) -> tuple[NodeT, ...]:
         """
         Returns the children of *node*.
         """
-        assert node in self
-
         return self._parent_to_children[node]
 
     @memoize_method
     def depth(self, node: NodeT) -> int:
         """
         Returns the depth of *node*, with the root having depth 0.
         """
-        assert node in self
-
         if self.is_root(node):
             # => None
             return 0
 
         parent_of_node = self.parent(node)
-        assert parent_of_node is not None
 
-        return 1 + self.depth(parent_of_node)
+        from typing import cast
 
-    def is_root(self, node: NodeT) -> bool:
-        assert node in self
+        # cast-reason: parent_of_node can not be None (as per is_root check)
+        return 1 + self.depth(cast(NodeT, parent_of_node))
```

Comment:
Use `not_none`?

Reply:
In that case, wouldn't it be easier just to leave the assert there, instead of the `not_none`?

Reply:
Would something like this work?
```python
parent_of_node = self.parent(node)
if parent_of_node is None:
	return 0

return 1 + self.depth(parent_of_node)
```
? No need to hide the None check in another function (which tricks mypy) + this only does the parent lookup once.

Reply:
That sounds good, thanks! [cdc3235](https://github.com/inducer/loopy/pull/887/commits/cdc32351ec0e84e8357c3b74cfe23835f2e92e9d)